### ISSUE TITLE
[3.7.0] Generic class with private property generates d.ts file missing type on generic property

### ISSUE THREAD

**Issue by fearthecowboy** [microsoft/TypeScript#33877]
**TypeScript Version:**  `3.7.0-beta ` (and just tested with typescript@next)

**Search Terms:** private field declaration

Given a class: 

``` ts
class foo<T> {
    private get value() : T {
        return this.v;
    }   
    constructor (private v:T) {
    }
}
```
**Actual behavior:**

running:
`.\node_modules\.bin\tsc .\foo.ts --target es2016 --declaration` 

you will get a `foo.d.ts` file that looks like this:

``` ts
declare class foo<T> {
    private v;
    private get value();
    constructor(v: T);
}
```

Dependent projects consuming that `d.ts` file will break, because the generic type isn't propagated to the private field declaration.

``` error
node_modules/foo/index.d.ts:3:17 - error TS7033: Property 'value' implicitly has type 'any', because its get accessor lacks a return type annotation.           
     private get value();                                                                                                                                                        ~~~~~           
```

**Expected behavior:**
``` ts
declare class foo<T> {
    private v: T;
    private get value() : T;
    constructor(v: T);
}
``` 

**Playground Link:**  playground doesn't generate d.ts file... (grrr!)

**Related Issues:** no.

**Comment by nmain** [comment]
Shouldn't the expected behavior be not to change the emit, but to remove the error?  `private` members are only emitted in declarations to guard against name collisions, and this doesn't produce an error even though `v` is untyped:

```ts
declare class C { private v }
```

**Comment by fearthecowboy** [comment]
O_o

Given that `.d.ts` files are typing info, if the error was removed (in 3.7+)  that would mean that consumers of the API that didn't have the latest typescript would break. 

That doesn't quite seem right.

**Comment by fearthecowboy** [comment]
That seems reasonable. `never` was introduced in what, 2.0?

I'm ok with that.

The tangential question is ... what is the expectation for `d.ts` files and consumability with prior compiler versions.

**Comment by ajafff** [comment]
The error when consuming such declaration files will be fixed by #33896

**Comment by sandersn** [comment]
The error should just be removed. Accessors were not allowed in d.ts files until 3.6 anyway, so it is not possible to make them compatible with anything 3.5 or below.

@ajaff Thanks for the PR!
